src/types.ts

```diff
@@ -1,25 +1,29 @@
 import { Static, Type } from '@sinclair/typebox';
 import { FastifyLoggerInstance } from 'fastify';
+import pino from 'pino';
 
-export interface PubSubConfig {
+export interface PubSubConfig<Data, Context> {
   /**
    * Handler
    */
-  handler: PubSubHandler;
+  handler: PubSubHandler<Data, Context>;
   /**
    * OnError Handler
    *
    * When this is set, errors will not be
    * thrown.
    */
   onError?: OnErrorHandler;
+
+  parser?: (data: unknown) => Data | Promise<Data>;
```

Comment:
Needs some docs, here, and probably in the readme too:

- What's the contract? Should it return data or null?, data or throw? Some kind of Either<data,error> type?
- What happens if the data is unparsable? Does the handler return ok or error? How can the user change that if neccessary?

Reply:
Not sure, the contract right now is return data or throw. What do you think?

The user can catch the exception with `onError` since the parser function is executed in the try-catch.